code/modules/antagonists/space_dragon/carp_rift.dm

```diff
@@ -0,0 +1,239 @@
+	name = "carp rift"
+	desc = "A rift akin to the ones space carp use to travel long distances."
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 100, BOMB = 50, BIO = 100, FIRE = 100, ACID = 100)
+	max_integrity = 300
+	icon = 'icons/obj/carp_rift.dmi'
+	icon_state = "carp_rift_carpspawn"
+	light_color = LIGHT_COLOR_PURPLE
+	light_range = 10
+	anchored = TRUE
+	density = FALSE
+	plane = MASSIVE_OBJ_PLANE
+	/// The amount of time the rift has charged for.
+	var/time_charged = 0
+	/// The maximum charge the rift can have.
+	var/max_charge = 300
+	/// How many carp spawns it has available.
+	var/carp_stored = 1
+	/// A reference to the Space Dragon antag that created it.
+	var/datum/antagonist/space_dragon/dragon
+	/// Current charge state of the rift.
+	var/charge_state = CHARGE_ONGOING
+	/// The interval for adding additional space carp spawns to the rift.
+	var/carp_interval = 60
+	/// The time since an extra carp was added to the ghost role spawning pool.
+	var/last_carp_inc = 0
+	/// A list of all the ckeys which have used this carp rift to spawn in as carps.
+	var/list/ckey_list = list()
+
+/obj/structure/carp_rift/Initialize(mapload)
+	. = ..()
+
+	AddComponent( \
+		/datum/component/aura_healing, \
+		range = 0, \
+		simple_heal = 5, \
+		limit_to_trait = TRAIT_HEALS_FROM_CARP_RIFTS, \
+		healing_color = COLOR_BLUE, \
+	)
+
+	START_PROCESSING(SSobj, src)
+
+// Carp rifts always take heavy explosion damage. Discourages the use of maxcaps
+// and favours more weaker explosives to destroy the portal
+// as they have the same effect on the portal.
+/obj/structure/carp_rift/ex_act(severity, target)
+	return ..(min(EXPLODE_HEAVY, severity))
+
+/obj/structure/carp_rift/examine(mob/user)
+	. = ..()
+	if(time_charged < max_charge)
+		. += span_notice("It seems to be [(time_charged / max_charge) * 100]% charged.")
+	else
+		. += span_warning("This one is fully charged. In this state, it is poised to transport a much larger amount of carp than normal.")
+
+	if(isobserver(user))
+		. += span_notice("It has [carp_stored] carp available to spawn as.")
+
+/obj/structure/carp_rift/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	playsound(src, 'sound/magic/lightningshock.ogg', 50, TRUE)
+
+/obj/structure/carp_rift/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	if(charge_state != CHARGE_COMPLETED)
+		if(dragon)
+			to_chat(dragon.owner.current, span_boldwarning("A rift has been destroyed! You have failed, and find yourself  weakened."))
+			dragon.destroy_rifts()
+	dragon = null
+	return ..()
+
+/obj/structure/carp_rift/process(delta_time)
+	// If we're fully charged, just start mass spawning carp and move around.
+	if(charge_state == CHARGE_COMPLETED)
+		if(DT_PROB(1.25, delta_time) && dragon)
+			var/mob/living/newcarp = new dragon.ai_to_spawn(loc)
+			newcarp.faction = dragon.owner.current.faction.Copy()
+		if(DT_PROB(1.5, delta_time))
+			var/rand_dir = pick(GLOB.cardinals)
+			Move(get_step(src, rand_dir), rand_dir)
```

Comment:
I know this is pre-your PR, but can we use ``move_to()`` instead, ``Move()`` generally shouldn't be directly called like this.